### PR TITLE
add astc support (Great Job)

### PR DESCRIPTION
**Pull request by weiwest**
support 2D LDR SRGB 
encode mode 4x4 8x8

### CODE REVIEW

File: cocos/renderer/CCTexture2D.cpp [hunk 1]
```diff
@@ -321,7 +323,8 @@ bool Texture2D::updateWithImage(Image* image, backend::PixelFormat format, int i
         updateWithMipmaps(image->getMipmaps(), image->getNumberOfMipmaps(), image->getPixelFormat(), renderFormat, imageWidth, imageHeight, image->hasPremultipliedAlpha(), index);
     }
     else if (image->isCompressed())
-    {
+    {   
+        CCLOG("weichao renderFormat %d,%d", renderFormat, image->getPixelFormat());
```
Comment (huizhen) [inline cocos/renderer/CCTexture2D.cpp]: Better remove the author name in the log.

**Comment by halx99** [comment]
Should remove author from log & fix travis ci compile error

File: cocos/base/astc.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    uint32_t  yblocks = (height + ydim - 1) / ydim;
+
+    imageblock pb;
+    for (uint32_t by = 0; by < yblocks; by++) {
+        for (uint32_t bx = 0; bx < xblocks; bx++) {
+
+            physical_compressed_block pcb = *(physical_compressed_block*)in;
+            symbolic_compressed_block scb;
+            physical_to_symbolic(xdim, ydim, 1, pcb, &scb);
+            decompress_symbolic_block(DECODE_LDR_SRGB, xdim, ydim, 1, bx * xdim, by * ydim, 0, &scb, &pb);
```
Comment (huizhen) [inline cocos/base/astc.cpp]: Where is the function's implementation, are u forgot to commit the include file and lib folder?

**Comment by halx99** [comment]
You should modify external/config.json to v27

**Comment by halx99** [comment]
[ci rebuild]

**Comment by halx99** [comment]
@weiwest still faild, the ios platform does't allow the c API 'system' to create child process.

**Comment by halx99** [comment]
@weiwest do you real require 'system' API?

**Comment by weiwest** [comment]
system use for read/write files
It`s not be used in engine
I have commented the relevant code.

**Comment by halx99** [comment]
looks good to me, merge it, thanks weitest provide so much excellent PR, then great job.

**Comment by huizhen** [comment]
Good job.

**Pull request merged**